napari/_qt/layers/qt_image_layer.py

```diff
@@ -56,55 +56,69 @@ class QtImageProperties(QtLayerProperties):
     def __init__(self, layer):
         super().__init__(layer)
 
-        self.layer.events.colormap.connect(self._on_colormap_change)
-        self.layer.events.interpolation.connect(self._on_interpolation_change)
-        self.layer.events.rendering.connect(self._on_rendering_change)
-
-        row = self.grid_layout.rowCount()
-        comboBox = QComboBox()
-        for cmap in self.layer.colormaps:
-            comboBox.addItem(cmap)
-        comboBox._allitems = set(self.layer.colormaps)
-        index = comboBox.findText(self.layer.colormap[0], Qt.MatchFixedString)
-        comboBox.setCurrentIndex(index)
-        comboBox.activated[str].connect(
-            lambda text=comboBox: self.changeColor(text)
-        )
-        self.grid_layout.addWidget(QLabel('colormap:'), row, self.name_column)
-        self.grid_layout.addWidget(comboBox, row, self.property_column)
-        self.colormap_combobox = comboBox
-
-        row = self.grid_layout.rowCount()
-        interp_comboBox = QComboBox()
-        for interp in Interpolation:
-            interp_comboBox.addItem(str(interp))
-        index = interp_comboBox.findText(
-            self.layer.interpolation, Qt.MatchFixedString
-        )
-        interp_comboBox.setCurrentIndex(index)
-        interp_comboBox.activated[str].connect(
-            lambda text=interp_comboBox: self.changeInterpolation(text)
-        )
-        self.interpComboBox = interp_comboBox
-        self.grid_layout.addWidget(
-            QLabel('interpolation:'), row, self.name_column
-        )
-        self.grid_layout.addWidget(interp_comboBox, row, self.property_column)
-
-        row = self.grid_layout.rowCount()
-        renderComboBox = QComboBox()
-        for render in Rendering:
-            renderComboBox.addItem(str(render))
-        index = renderComboBox.findText(
-            self.layer.rendering, Qt.MatchFixedString
-        )
-        renderComboBox.setCurrentIndex(index)
-        renderComboBox.activated[str].connect(
-            lambda text=renderComboBox: self.changeRendering(text)
-        )
-        self.renderComboBox = renderComboBox
-        self.grid_layout.addWidget(QLabel('rendering:'), row, self.name_column)
-        self.grid_layout.addWidget(renderComboBox, row, self.property_column)
+        if hasattr(self.layer, 'colormap'):
```

Comment:
i think all of this `hasattr` checking is really strange

if you are having trouble with inheriting things, just take the base code you want shared everywhere and create a new class that both `Image` and `Surface` can inherit from

Reply:
ok, makes sense - i can actually reduce some code duplication between with the labels layer properties here too

Reply:
actually the labels layer properties doesn't have the duplicate code. I just worry that we'll now have extra classes. I'll think about how to do this nicely

Reply:
done